fastdata/core.py

```diff
@@ -44,23 +102,123 @@ def generate(self,
+            token (Optional[str], optional): The token to use to commit to the repo. Defaults to the token saved on the machine.
+            delete_files_after (bool, optional): Whether to delete files after processing. Defaults to True.
+
+        Returns:
+            tuple[str, list[dict]]: A tuple with the generated repo_id and the list of generated data dictionaries.
+        """
+        dataset_dir = Path(repo_id.replace("/", "_"))
+        dataset_dir.mkdir(parents=True, exist_ok=True)
+        data_dir = dataset_dir / "data"
+        data_dir.mkdir(exist_ok=True)
+
+        try:
+            scheduler = CommitScheduler(
+                repo_id=repo_id,
+                repo_type="dataset",
+                folder_path=dataset_dir,
+                every=commit_every,
+                private=private,
+                token=token,
+            )
+            DatasetCard(
+                DATASET_CARD_TEMPLATE.format(
+                    title=repo_id,
+                    model_id=self.cli.model,
+                    system_prompt=sp,
+                    prompt_template=prompt_template,
+                    sample_input=inputs[:2],
+                )
+            ).save(dataset_dir / "README.md")
+
+            results = []
+            total_inputs = len(inputs)
+
+            with tqdm(total=total_inputs) as pbar:
+                with concurrent.futures.ThreadPoolExecutor(
+                    max_workers=max_workers
+                ) as executor:
+                    futures = [
+                        executor.submit(
+                            self._process_input,
+                            prompt_template,
+                            schema,
+                            temp,
+                            sp,
+                            input_data,
+                        )
+                        for input_data in inputs
+                    ]
+                    completed = 0
+
+                    for completed_future in concurrent.futures.as_completed(futures):
+                        result = completed_future.result()
+                        if result is not None:
+                            results.append(result)
+                        completed += 1
+                        pbar.update(1)
+
+                        if completed % save_interval == 0 or completed == total_inputs:
+                            if results:
+                                with scheduler.lock:
+                                    self._save_results(
+                                        results, data_dir / f"train-{uuid4()}.jsonl"
+                                    )
+                                results.clear()
+                    scheduler._push_to_hub()
+                    scheduler.stop()
+        except Exception as e:
+            raise e
+        finally:
+            if delete_files_after:
+                shutil.rmtree(dataset_dir)
```

Comment:
```suggestion
        finally:
            if delete_files_after:
                shutil.rmtree(dataset_dir)
```

(nit) "raise" is implicit when using `finally`

(example:
```py
try:
    1 / 0
finally:
    print("here")
```
=> it prints "here" and then raises the exception)